### PR TITLE
TRIZ-3256: Validate incoming request with JSON schema

### PR DESCRIPTION
**Pull request by johntowell**
* Update account/create request schema and example
   * Update account/data request schema

### CODE REVIEW

**Comment by johntowell** [comment]
@abuecker @eweise @antoinemeyer @LowellBateman 

I think this will break some tests currently, I wont merge until all the e2e tests pass.

**Comment by johntowell** [comment]
Cool this is working locally, ready to merge. 

~/projects/accelerator (master *)$ wdio wdio.conf.js 

account type
  ✓ should have empty labels when nothing selected (197ms)
  ✓ should have first row visible (512ms)
  ✓ should have second row visible (450ms)
joint account
  ✓ should not be visible by default (214ms)
  ✓ should be visible when selected in combo (361ms)
Login
  ✓ should be able to login (582ms)
  ✓ should be able to logout (1445ms)
Onboarding
  ✓ should be able to navigate through onboarding (4125ms)
Portfolio
  ✓ should have a chart (32ms)
Signup mailing address checkbox
  ✓ should clear out when checked (553ms)
Signup previous address checkbox
  ✓ should not show previous home address by default (467ms)
  ✓ should show previous home address when checked (497ms)
SignUp
  ✓ should have state dropdown (697ms)


13 passing (18.50s)

File: .eslintrc [hunk 1]
```diff
@@ -129,7 +129,7 @@
 /**
  * Style
  */
-    "max-len": [2, 80, 2],           // http://eslint.org/docs/rules/max-len.html
+    "max-len": [2, 120, 2],           // http://eslint.org/docs/rules/max-len.html
```
Comment (abuecker) [inline .eslintrc]: Let's all standardize on the traditional 80

File: server/api.js [hunk 2]
```diff
@@ -40,6 +52,7 @@ export default function api(server) {
       // content
       if (/account\/file/.test(reqToMock.uri)) {
         server[reqToMock.method](reqToMock.uri, function(req, res, next) {
+          validateRequest(reqToMock, req, res, next);
```
Comment (abuecker) [inline server/api.js]: `/*/account/file` only has a GET method.  Is this needed?

File: server/api.js [hunk 3]
```diff
@@ -91,6 +105,7 @@ export default function api(server) {
        */
       if (/account\/data\/periods/.test(reqToMock.uri)) {
         server[reqToMock.method](reqToMock.uri, function(req, res, next) {
+          validateRequest(reqToMock, req, res, next);
```
Comment (abuecker) [inline server/api.js]: This route only has a GET method:

`Mock: /v1/account/data/periods GET`

File: .eslintrc (hunk 1 above)
Comment (johntowell) [inline .eslintrc]: ewwww, really?

Comment (antoinemeyer) [inline .eslintrc]: our java code style is 160

Comment (abuecker) [inline .eslintrc]: think you forgot a trailing 0 there @antoinemeyer

Comment (antoinemeyer) [inline .eslintrc]: we'll get you a bigger screen ;)

Comment (abuecker) [inline .eslintrc]: I'm guessing you haven't spent to much time shelled into a server debugging a problem.  Long lines are killer.

This eslint config is copy/paste from AirBnB's which has been widely adopted.

**Comment by johntowell** [comment]
80 columns it is :)

**Comment by antoinemeyer** [comment]
please don't change the java one ! :p

**Comment by johntowell** [comment]
you can't even declare your method name with 80 characters in java!

**Comment by eweise** [comment]
I think our java standard was 160 columns

**Comment by abuecker** [comment]
:+1:

**Pull request merged**